### PR TITLE
Update create_and_fill_tensor to work with InputGenerateData

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Create new InputGenerateData struct with tensor generation params
 - Update create_and_fill_tensor to work with InputGenerateData

### Tickets:
 - [CVS-117139](https://jira.devtools.intel.com/browse/CVS-117139)

### CODE REVIEW

**Comment by olpipi** [comment]
updated

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/adaptive_pooling.cpp [hunk 1]
```diff
@@ -162,7 +162,9 @@ class AdaPoolLayerCPUTest : public testing::WithParamInterface<AdaPoolLayerCPUTe
                     dataPtr[i] = pooledVector[i];
                 }
             } else {
-                    tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 2560, 0, 256);
+                    tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(),
```
Comment (iefode) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/adaptive_pooling.cpp:165]: to use `InputGenerateData` arg as an  argument of `create_and_fill_tensor` to avoid of misusing of the structure?

Comment (olpipi) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/adaptive_pooling.cpp:165]: You mean use something like that?
```
InputGenerateData in_data;
in_data.range = 2560;
in_data.start_from = 0;
...
tensor = ov::test::utils::create_and_fill_tensor(..., in_data);
```
I will take several days to update all usages of create_and_fill_tensor like that

Comment (iefode) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/adaptive_pooling.cpp:165]: Yeah, correct.
It's not burning PR

Comment (olpipi) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/adaptive_pooling.cpp]: Unfortunately I cannot finish builders reworking (makeConstant) without this PR.
So I updated all...

File: src/tests/functional/plugin/shared/src/snippets/mha.cpp [hunk 2]
```diff
@@ -81,9 +81,16 @@ void MHA::generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticSha
...
         ov::Tensor tensor;
+        ov::test::utils::InputGenerateData in_data;
+<<<<<<< HEAD
         // To avoid big relative errors in the vicinity of zero, only positive values are generated for bf16 precision
-        int start_from = model_input.get_element_type() == ov::element::bf16 ? 0 : -1;
-        tensor = ov::test::utils::create_and_fill_tensor(model_input.get_element_type(), model_input.get_shape(), 2, start_from, 256);
+        in_data.start_from = model_input.get_element_type() == ov::element::bf16 ? 0 : -1;
+=======
+        in_data.start_from = -1;
+>>>>>>> d89271e8c1 (Apply comments)
```
Comment (v-Golubev) [inline src/tests/functional/plugin/shared/src/snippets/mha.cpp:90]: Please do not overwrite the [recent fix on input generation for bf16 MHA test cases](https://github.com/openvinotoolkit/openvino/pull/21250).

Comment (olpipi) [reply src/tests/functional/plugin/shared/src/snippets/mha.cpp:90]: Thanks for the comment.
This is probably rebase issue. I manually kept your fix. But I don't see it now.

Comment (olpipi) [reply src/tests/functional/plugin/shared/src/snippets/mha.cpp]: Updated. Your fix is kept

File: src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@ ov_add_test_target(
             PUBLIC
                 conformance_shared
                 op_conformance_utils
+                sharedTestClasses
```
Comment (vurusovs) [inline src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/CMakeLists.txt:18]: Why this is required?

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp [hunk 4]
```diff
@@ -168,12 +168,15 @@ void fill_data_roi(ov::runtime::Tensor& tensor,
...
+// k_range = start_from * k have to be integer value
+// k_start = range      * k have to be integer value
+// otherwise UB. ex 0.35 * 10 = 3.5
 template <class T>
 void inline fill_data_random(T* pointer,
                              std::size_t size,
-                             const uint32_t range = 10,
-                             double_t start_from = 0,
-                             const int32_t k = 1,
+                             const double range = 10,
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:177]: Why the change is required? Note that strict comparison `range == 0` won't work for floating types

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/tests/test_utils/common_test_utils/src/postgres_link.cpp [hunk 5]
```diff
@@ -557,12 +557,12 @@ class PostgreSQLEventListener : public ::testing::EmptyTestEventListener {
...
              << " WHERE sr_id=" << this->testSuiteId;
         auto pgresult = connectionKeeper->query(sstr.str().c_str(), PGRES_COMMAND_OK);
-        CHECK_PGRESULT(pgresult, "Cannot update test suite results", return);
+        CHECK_PGRESULT(pgresult, "Cannot update test suite results", return );
         this->testSuiteId = 0;
 
         if (reportingLevel == REPORT_LVL_FAST) {
             pgresult = connectionKeeper->query(joinedQuery.str().c_str());
-            CHECK_PGRESULT(pgresult, "Cannot update test cases results", return);
+            CHECK_PGRESULT(pgresult, "Cannot update test cases results", return );
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/src/postgres_link.cpp:565]: Why? It's not aligned with code style

File: src/tests/test_utils/common_test_utils/src/postgres_helpers.cpp [hunk 6]
```diff
@@ -28,12 +28,12 @@ fnPQresultErrorMessage PQresultErrorMessage;
 
 const char* PGPrefix(const char* text, ::testing::internal::GTestColor color) {
 #ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-security"
+#    pragma GCC diagnostic push
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/src/postgres_helpers.cpp:31]: Why?

File: src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp [hunk 7]
```diff
@@ -59,6 +58,17 @@ ov::Tensor create_and_fill_tensor(const ov::element::Type element_type,
     return tensor;
 }
 
+ov::Tensor create_and_fill_tensor(const ov::element::Type element_type,
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp:61]: mark it as `todo`  to remove the function in the future

File: src/tests/test_utils/common_test_utils/include/common_test_utils/ov_tensor_utils.hpp [hunk 8]
```diff
@@ -9,9 +9,27 @@
...
+          resolution(_resolution),
+          seed(_seed){};
+};
+
+ov::Tensor create_and_fill_tensor(const ov::element::Type element_type,
+                                  const ov::Shape& shape,
+                                  const InputGenerateData& inGenData = InputGenerateData(0, 10, 1, 1));
+
+// Legacy impl for contrig repo
 ov::Tensor create_and_fill_tensor(const ov::element::Type element_type,
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/ov_tensor_utils.hpp:30]: to remove or mark as todo?

File: src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/CMakeLists.txt (hunk 3 above)
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/CMakeLists.txt:18]: this is duplication, because of the lib is linked via `conformance_shared`

Comment (olpipi) [reply src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/CMakeLists.txt]: fixed

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp (hunk 4 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:177]: Oh... thanks. I missed that
We need to change type to support not integer intervals. E.g. [0.1, 0.2]

File: src/tests/test_utils/common_test_utils/src/postgres_link.cpp (hunk 5 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/postgres_link.cpp]: clang-format changed it automatically. reverted

File: src/tests/test_utils/common_test_utils/src/postgres_helpers.cpp (hunk 6 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/postgres_helpers.cpp]: clang-format changed it automatically. reverted

File: src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp (hunk 7 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp:63]: done

File: src/tests/test_utils/common_test_utils/include/common_test_utils/ov_tensor_utils.hpp (hunk 8 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/ov_tensor_utils.hpp:31]: marked as todo

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp (hunk 4 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:177]: Comparison with std::numeric_limits<float>::epsilon() is enough here

**Comment by olpipi** [comment]
Last rebase is performed with errors. Rebased it again with fixes

**Review by vurusovs — APPROVED** [review]

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 9]
```diff
@@ -93,7 +93,10 @@ TEST_P(OVInferRequestDynamicTests, InferDynamicNetwork) {
     ov::InferRequest req;
     const std::string outputname = function->outputs().back().get_any_name();
     for (auto& shape : vectorShapes) {
-        ov::runtime::Tensor inTensor = ov::test::utils::create_and_fill_tensor(element::f32, shape, 100, -50);
+        ov::test::utils::InputGenerateData in_data;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:96]: to set in constructor?

File: src/tests/functional/plugin/shared/src/snippets/convert.cpp [hunk 10]
```diff
@@ -86,8 +86,11 @@ void Convert::generate_inputs(const std::vector<ov::Shape>& targetInputStaticSha
         ov::Tensor tensor;
         int32_t startFrom, range, resolution;
         std::tie(startFrom, range, resolution) = params[i];
-        tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i],
-                                                         range, startFrom, resolution);
+        ov::test::utils::InputGenerateData in_data;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/snippets/convert.cpp:89]: to use constructor instead of setting structure fields?
